peps/pep-0752.rst

```diff
@@ -254,33 +262,39 @@ organizations have no special permissions and are essentially only public.
 Root grants given to `community projects <grant-approval-criteria_>`_ SHALL
 always be public.
 
+When a `child grant <child-grant_>`_ is created, its public status SHALL be
+inherited from the `root grant <root-grant_>`_. Owners of child grants MAY
+make them public at any time. If a grant is public, it MUST NOT be made private
+unless the owner of the grant is the owner of every project that matches the
+namespace.
+
 .. _repository-metadata:
 
 Repository Metadata
 -------------------
 
-To allow installers and other tooling insight into this metadata for a given
-artifact upload of a namespaced package, the :pep:`JSON API <691>` MUST include
-the following keys:
-
-* ``namespace``: This is the associated `normalized <naming_>`_
-  namespace e.g. ``foo-bar``. If the namespace matches a child grant and the
-  user happens to be authorized for both the child and the root grant, this
-  MUST be the namespace associated with the child grant.
-* ``owner``: This is the organization with which the user is associated and
-  owner of the grant. If the namespace is `public <public-namespaces_>`_ and
-  the user is not part of a `permitted <grant-ownership_>`_ organization, this
-  key MUST be set to ``__public__``. This is useful for tools that wish to make
-  a distinction between official and community packages.
+To allow installers and other tooling insight into this project-level metadata
+of a namespaced project, the :pep:`JSON API <691>` version will be incremented
+and support new keys for the project endpoint.
 
-The `Simple API`__ MAY include the aforementioned keys as attributes, for
-example:
+The ``owner`` key SHOULD be added and refer to the owner of the project,
+whether an organization or a user.
 
-__ https://packaging.python.org/en/latest/specifications/simple-repository-api/#base-html-api
+The ``namespace`` key MAY be added and MUST be ``null`` if the project does not
+match an active namespace grant. If the project does match a namespace grant,
+the value MUST be a mapping with the following keys:
 
-.. code-block:: html
+* ``name``: This is the associated `normalized <naming_>`_ namespace e.g.
+  ``foo-bar``. If the owner of the project owns multiple matching grants then
```

Comment:
I don't understand the rationale for "the most number of characters" requirement.

Reply:
I tried to make this case more explicit: https://github.com/python/peps/pull/3903#discussion_r1715724787

Do you have a suggestion for a better wording?

Reply:
No suggestion for better wording, but suggest adding an elaboration, turning @AA-Turner 's question into a statement.  You could add an asterisk or something and add a little text after the bullet list with the given example.